sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultClient.java

```diff
@@ -0,0 +1,244 @@
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import static java.util.logging.Level.INFO;
+import static java.util.logging.Level.WARNING;
+
+/**
+ * The REST client specific to Azure KeyVault.
+ */
+class KeyVaultClient extends DelegateRestClient {
+
+    /**
+     * Stores the logger.
+     */
+    private static final Logger LOGGER = Logger.getLogger(KeyVaultClient.class.getName());
+
+    /**
+     * Stores the API version postfix.
+     */
+    private static final String API_VERSION_POSTFIX = "?api-version=7.1";
+
+    /**
+     * Stores the KeyVault URI.
+     */
+    private final String keyVaultUri;
+
+    /**
+     * Stores the tenant ID.
+     */
+    private String tenantId;
+
+    /**
+     * Stores the client ID.
+     */
+    private String clientId;
+
+    /**
+     * Stores the client secret.
+     */
+    private String clientSecret;
+
+    /**
+     * Constructor.
+     *
+     * @param keyVaultUri the KeyVault URI.
+     */
+    KeyVaultClient(String keyVaultUri) {
+        super(RestClientFactory.createClient());
+        LOGGER.log(INFO, "Using KeyVault: {0}", keyVaultUri);
+        if (!keyVaultUri.endsWith("/")) {
+            keyVaultUri = keyVaultUri + "/";
+        }
+        this.keyVaultUri = keyVaultUri;
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param keyVaultUri the KeyVault URI.
+     * @param tenantId the tenant ID.
+     * @param clientId the client ID.
+     * @param clientSecret the client secret.
+     */
+    KeyVaultClient(final String keyVaultUri, final String tenantId, final String clientId, final String clientSecret) {
+        this(keyVaultUri);
+        this.tenantId = tenantId;
+        this.clientId = clientId;
+        this.clientSecret = clientSecret;
+    }
+
+    /**
+     * Get the access token.
+     *
+     * @return the access token.
+     */
+    private String getAccessToken() {
+        LOGGER.entering("KeyVaultClient", "getAccessToken");
+        String accessToken = null;
+        try {
+            AuthClient authClient = new AuthClient();
+            String resource = URLEncoder.encode("https://vault.azure.net", "UTF-8");
+            if (tenantId != null && clientId != null && clientSecret != null) {
+                accessToken = authClient.getAccessToken(resource, tenantId, clientId, clientSecret);
+            } else {
+                accessToken = authClient.getAccessToken(resource);
+            }
+        } catch (UnsupportedEncodingException uee) {
+            LOGGER.log(WARNING, "Unsupported encoding", uee);
+        }
+        LOGGER.exiting("KeyVaultClient", "getAccessToken", accessToken);
+        return accessToken;
+    }
+
+    /**
+     * Get the list of aliases.
+     *
+     * @return the list of aliases.
+     */
+    public List<String> getAliases() {
```

Comment:
Don't need to have public methods on a package private class

Reply:
Fixed

Reply:
Fixed